### DISCUSSION TITLE
Edit Layers - Finish - how to define an event?

### DISCUSSION THREAD

**Discussion by r4naf** [geoman-io/leaflet-geoman#847] [Questions & Help]
help me, how to track the event when the button is pressed? Thank!
![pic-00](https://user-images.githubusercontent.com/37614712/110080694-e2003780-7d9b-11eb-987b-a480c4836a2b.png)

**Comment by Falke-Design** [comment]
You can use the event `pm:actionclick` but if you want to know when the Edit Mode is disabled I suggest you to use the event `pm:globaleditmodetoggled`:

```
map.on('pm:globaleditmodetoggled', (e)=>{
   if(e.enabled === false){
      // Your code
   }
});
```

https://github.com/geoman-io/leaflet-geoman#actions
https://github.com/geoman-io/leaflet-geoman#edit-mode

**Comment by r4naf** [comment]
Thank!!! Tell me how to find out which polygon has been updated?

```
map.on('pm:globaleditmodetoggled', (e)=>{
		if(e.enabled === false){
			e.layer.on('pm:update', ({ layer }) => {
				/* ? */
				console.log('This landfill has been updated.', layer);
				/* ? */
			})
	   }
});
```

**Comment by Falke-Design** [comment]
you have no reference to the layer in the event `pm:globaleditmodetoggled`, use this:

```
map.on('pm:create',(e)=>{
// add event to layer when it is drawn
  e.layer.on('pm:update',(x)=>{
    console.log(x);
  })
})
```

**Comment by r4naf** [comment]
> you have no reference to the layer in the event `pm:globaleditmodetoggled`, use this:
> 
> ```
> map.on('pm:create',(e)=>{
> // add event to layer when it is drawn
>   e.layer.on('pm:update',(x)=>{
>     console.log(x);
>   })
> })
> ```

Unfortunately, this does not work if I have loaded the polygons from the database.
This works - if polygons were created and modified.
And if the polygons were counted from the database (ajax, geojson ...), then for these polygons, the event does not occur.

**Comment by Falke-Design** [comment]
You add the the polygons anywhere to the map and there you need to add the listener `pm:update` to the layer. If you don't understand, please share your code as snippet, online demo or as `.txt` file

**Comment by r4naf** [comment]
> You add the the polygons anywhere to the map and there you need to add the listener `pm:update` to the layer. If you don't understand, please share your code as snippet, online demo or as `.txt` file

```
	var polygonLayer = L.geoJSON(null, {
			polygonToLayer: function(feature,latlng){
				return new L.polygon(latlng, {
				  title: '111'
				})
			}
		});
	
	
	function ShowPolygon() {
		$.ajax({
				url: "getpoligon.php", 
				type: "POST",
				data: {},
				async: false,
				success: function(result){
						if (result.features !== null){
							polygonLayer.addData(result);
							layerPoligonGroup.addLayer(polygonLayer);
						}
					}
				}
		);
	};

ShowPolygon();
```

**Comment by Falke-Design** [comment]
Add to to `L.geoJSON` the event:

```
var polygonLayer = L.geoJSON(null, {
			polygonToLayer: function(feature,latlng){
				return new L.polygon(latlng, {
				  title: '111'
				})
			}
		});

polygonLayer.on('pm:update',(x)=>{
      console.log(x);
    })

function ShowPolygon() {
...

```

**Comment by r4naf** [comment]
Falke-Design, thank you friend! Things are good! I like **leaflet-geoman**. This is the best library for editing polygons!

> Add to to `L.geoJSON` the event:
